fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/FieldKeyExpression.java

```diff
@@ -318,12 +320,14 @@ public boolean equals(Object o) {
         }
 
         FieldKeyExpression that = (FieldKeyExpression)o;
-        return this.fieldName.equals(that.fieldName) && (this.fanType == that.fanType);
+        return this.fieldName.equals(that.fieldName) &&
+               this.fanType == that.fanType &&
+               this.nullStandin == that.nullStandin;
     }
 
     @Override
     public int hashCode() {
-        return fieldName.hashCode() + fanType.hashCode();
+        return fieldName.hashCode() + fanType.hashCode() + nullStandin.hashCode();
```

Comment:
Yes, that was intentional.

> Make FieldKeyExpression.equals and hashCode take nullStandin into account. Keep planHash the same so as not to be disruptive.

Reply:
Perhaps I’d need to think about what the implications are, but it does seem unlikely that an actual plan would choose to flip this in the middle. I think my preference would be to update it so that the value is different for different `NullStandIn` values as that seems more safe, but maybe I’m wrong.